seismostats/analysis/estimate_beta.py

```diff
@@ -307,6 +389,62 @@ def estimate_b_positive(
         return out
 
 
+def make_more_incomplete(
+    magnitudes: np.ndarray,
+    times: dt.datetime,
+    delta_t: np.timedelta64 = np.timedelta64(60, "s"),
+    return_idx: bool = False,
+) -> tuple[np.ndarray, np.ndarray]:
+    """Return filtered magnitudes and times. Filter the magnitudes and times in
+    the following way: If an earthquake is smaller than the previous one and
+    less than ``delta_t`` away, the earthquake is removed.
+
+    Source:
+        E. Lippiello and G. Petrillo. Journal of Geophysical Research: Solid
+-       Earth, 129(2):e2023JB027849, 2024.
+
+    Args:
+        magnitudes: array of magnitudes, sorted in time (first
+                entry is the earliest earthquake).
+        times:      array of datetime objects of occurrence of each earthquake
+        delta_t:    time window in seconds to filter out events. default is 60
+                seconds.
+        return_idx: if True the indices of the events that were removed are
+                returned
+
+    Returns:
+        magnitudes: filtered array of magnitudes
+        times:      filtered array of datetime objects
+        idx:        indices of the events that were removed
+
+    """
+
+    # sort magnitudes in time
+    idx_sort = np.argsort(times)
+    magnitudes = magnitudes[idx_sort]
+    times = times[idx_sort]
+
+    idx_del = []
+    for ii in range(1, len(magnitudes)):
+        # get all times that are closer than delta_t
+        idx_close = np.where(times[ii] - times[:ii] < delta_t)[0]
```

Comment:
idx_close = times[ii] - times[:ii] < delta_t

Reply:
Not implemented, as we actually need the index. It might be ugly, but for now it works.